### DISCUSSION TITLE
How can tower-lsp get the buffer from editor

### DISCUSSION THREAD

**Discussion by Decodetalkers** [ebkalderon/tower-lsp#346] [Q&A] [answered]
![unknown](https://user-images.githubusercontent.com/60290287/174816074-5c30f7ec-803f-4e29-a1da-be66ffae88c5.png)
In javascript, I can use getText() get the text from editor,but I do not know hao can tower-lsp do that

**Comment by ratmice** [comment, marked as answer]
See the [did_change](https://github.com/ebkalderon/tower-lsp/blob/057b4aa91419aedf243f60efc56f14bef07a1a0b/examples/stdio.rs#L92-L96) function here in the example, the unbound parameter with the type [`DidChangeTextDocumentParams`](https://docs.rs/lsp-types/0.93.0/lsp_types/struct.DidChangeTextDocumentParams.html) has a field `content_changes`, I'm not familiar with the javascript API though everything you need to replicate it should be available from those.

**Reply by Decodetalkers** [reply]
> 

How ever , I found content_changes can only get the changed text. I want the full content of a text..

**Reply by ratmice** [reply]
you'll want to modify the following lines, to instead pass 
[TextDocumentSyncKind::FULL](https://docs.rs/lsp-types/0.93.0/lsp_types/struct.TextDocumentSyncKind.html) https://github.com/ebkalderon/tower-lsp/blob/057b4aa91419aedf243f60efc56f14bef07a1a0b/examples/stdio.rs#L17-L19

**Reply by Decodetalkers** [reply]
Very thanks!

**Reply by ebkalderon** [reply]
+1 to the recommendation to using `TextDocumentSyncKind::FULL` to receive the full text document on every sync! Is this a suitable answer to you, @Decodetalkers? Is it safe to mark this discussion as solved?

**Reply by Decodetalkers** [reply]
> +1 to the recommendation to using `TextDocumentSyncKind::FULL` to receive the full text document on every sync! Is this a suitable answer to you, @Decodetalkers? Is it safe to mark this discussion as solved?

it is solved, thanks!

**Reply by ebkalderon** [reply]
Terrific! Relieved to hear it. :smile:
